app/models/repo.rb

```diff
@@ -25,12 +25,11 @@ def generate_sparkline_data
           2000
         )
 
-        chart_builder = ChartBuilder.new(benchmark_runs.sort_by do |benchmark_run|
-          benchmark_run.initiator.created_at
-        end)
+        runs = benchmark_runs.sort_by { |run| run.initiator.created_at }
+        chart_builder = ChartBuilder.new(runs, benchmark_result_type).build_columns
 
         charts[benchmark_type.category] ||= []
-        charts[benchmark_type.category] << [benchmark_result_type.name, chart_builder.build_columns]
+        charts[benchmark_type.category] << [benchmark_result_type.name, { columns: chart_builder.columns.to_json, categories: chart_builder.categories }]
```

Comment:
It doesn't sound right that the Ruby class `Repo` returns data structure serialized in JSON. If you need it to be JSON on the client side, it's controller business to serialize data into JSON.